### PR TITLE
adding readme.nodejs.md for automated node sdk generation

### PR DESCRIPTION
**Pull request by schaabs**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by lmazuel — APPROVED** [review]
Syntax is perfect, I can't judge the NodeJS content.

File: specification/keyvault/resource-manager/readme.nodejs.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Node
+
+These settings apply only when `--nodejs` is specified on the command line.
+Please also specify `--node-sdks-folder=<path to the root directory of your azure-sdk-for-node clone>`.
+
+``` yaml $(nodejs)
+nodejs:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  payload-flattening-threshold: 2
```
Comment (amarzavery) [inline specification/keyvault/resource-manager/readme.nodejs.md]: I do not use this setting while generating the node sdk. Here is the output when I generate this locally from the node sdk gulp file.js

```
>>>>>>>>>>>>>>>>>>>Start: "azure-keyvault" >>>>>>>>>>>>>>>>>>>>>>>>>
Executing command:
------------------------------------------------------------
autorest --output-folder=/Users/amarz/sdk/azure-sdk-for-node/lib/services/keyvault/lib --package-name=azure-keyvault --nodejs --license-header=MICROSOFT_MIT_NO_VERSION  --azure-arm   https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/keyvault/data-plane/readme.md 
------------------------------------------------------------
Output:
AutoRest code generation utility [version: 2.0.4245; node: v8.9.1]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/private/var/folders/23/8g7f__gs1v39mrr49b8pc8pc0000gn/T/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4245)
   Installing AutoRest extension '@microsoft.azure/autorest.nodejs' (~2.1.25)
   Installing AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.44)

>>>>>>>>>>>>>>>>>>>>>End: "azure-keyvault" >>>>>>>>>>>>>>>>>>>>>>>>>


>>>>>>>>>>>>>>>>>>>Start: "azure-arm-keyvault" >>>>>>>>>>>>>>>>>>>>>>>>>
Executing command:
------------------------------------------------------------
autorest --output-folder=/Users/amarz/sdk/azure-sdk-for-node/lib/services/keyVaultManagement/lib --package-name=azure-arm-keyvault --nodejs --license-header=MICROSOFT_MIT_NO_VERSION  --azure-arm   https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/keyvault/resource-manager/readme.md 
------------------------------------------------------------
Output:
AutoRest code generation utility [version: 2.0.4245; node: v8.9.1]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/private/var/folders/23/8g7f__gs1v39mrr49b8pc8pc0000gn/T/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4245)
   Loading AutoRest extension '@microsoft.azure/autorest.nodejs' (~2.1.25->2.1.28)
   Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.44->2.3.44)

>>>>>>>>>>>>>>>>>>>>>End: "azure-arm-keyvault" >>>>>>>>>>>>>>>>>>>>>>>>>
```

**Comment by salameer** [comment]
Hi @schaabs !

THanks for sending this, Dan and Riki "assigned" are planning on plugging the whole Node SDK into the AUto Generation and will include this part in. @daschult and @RikkiGibson let's sync-up on this tomorrow and see if we should pull this in or wait for your upcoming changes.

**Comment by RikkiGibson** [comment]
I think this is a good starting point but config might still need to be tweaked. Will take a shot at doing the config changes that still appear to be necessary, and then pull in @lmazuel and @marstr for review and to see if we can get the bot to open some test PRs to azure-sdk-for-node.

**Pull request closed without merging**